Source/Others/DDMathEvaluator+FORM.m

```diff
@@ -38,7 +38,9 @@ + (NSDictionary *)hyp_directoryFunctions {
 
             otherValue = (variables[expression.variable]) ?: expression.variable;
 
-            if (![baseValue isEqualToString:otherValue]) {
+            BOOL baseValueIsEqual = ([baseValue isKindOfClass:[NSString class]] &&
+                                     ![baseValue isEqualToString:otherValue]);
```

Comment:
You don't need `equals` to support numbers. You could do `eq($age,15)` or just `$age == 15` (I think).